tests/core_tests/test_api.py

```diff
@@ -505,6 +505,55 @@ def test_project_path_property(self):
         self.assertEqual(self.tk.project_path, self.project_root)
 
 
+class TestDefaultStorageRootHook(TankTestBase):
+    """
+    Test the custom implementation of the default_storage_root hook
+    which sets the default root to a project-specific root.
+    """
+
+    @patch(
+        "sgtk.pipelineconfig.PipelineConfiguration.get_core_hooks_location",
+        return_value=os.path.join(
+            os.path.dirname(__file__), "test_api_data/example_storage_root_hook_1"
+        ),
+    )
+    def test_roots_with_custom_project_field(self, *mocks):
+        """
+        Project-specific root is retrieved from custom ShotGrid project field.
+        """
+        with patch.object(
+            self.tk.shotgun,
+            "find_one",
+            return_value={
+                "type": "Project",
+                "id": 1,
+                "sg_storage_root_name": "project_specific_root",
```

Comment:
Can we have another test (or another block in the same function) where we check the behavior when the API does not return a `sg_storage_root_name` field?

Also, is there a way to test the case where the `sg_storage_root_name` field is filled but the name does not exist on the SG File Location database (the issue we struggled with back in June)?